### ISSUE TITLE
socket decorator not working

### ISSUE THREAD

**Issue by cmanrique-surtec** [serhiisol/node-decorators#147]
I'm get this error:
```
\node_modules\.pnpm\@decorators+di@2.0.0\node_modules\@decorators\di\src\container.ts:43
      throw new MissingProviderError(injectable);

MissingProviderError: Missing provider myclass
at Function.get (\node_modules\.pnpm\@decorators+di@2.0.0\node_modules\@decorators\di\src\container.ts:43:13)       
    at <anonymous> (\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\socket.ts:31:49)
    at Generator.next (<anonymous>)
    at \node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\lib\src\socket.js:2:775
    at new Promise (<anonymous>)
    at __awaiter (\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\lib\src\socket.js:2:335)
    at attachController (\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\socket.ts:29:19)
    at <anonymous> (node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\socket.ts:21:45)
    at Array.forEach (<anonymous>)
    at attachControllers (\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\socket.ts:21:15)

Node.js v18.12.1
```

and  if add @Injectable to myclass
```
\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\middleware.ts:83
    if (middleware.length === 0) {
                   ^


TypeError: Cannot read properties of undefined (reading 'length')
    at <anonymous> (\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\middleware.ts:83:20)
    at new Promise (<anonymous>)
    at executeMiddleware (\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\middleware.ts:82:10)
    at Array.<anonymous> (\node_modules\.pnpm\@decorators+socket@3.4.0_mvchsu4374pbmjsjwdwg5vsvti\node_modules\@decorators\socket\src\socket.ts:39:5)
    at run (\node_modules\.pnpm\socket.io@4.6.1_3cxu5zja4e2r5wmvge7mdcljwq\node_modules\socket.io\dist\namespace.js:124:18)
    at Namespace.run (\node_modules\.pnpm\socket.io@4.6.1_3cxu5zja4e2r5wmvge7mdcljwq\node_modules\socket.io\dist\namespace.js:135:9)
    at Namespace._add (\node_modules\.pnpm\socket.io@4.6.1_3cxu5zja4e2r5wmvge7mdcljwq\node_modules\socket.io\dist\namespace.js:211:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v18.12.1
```

**Comment by serhiisol** [comment]
Could you provide a bit detailed example how you use it and setup?

**Comment by serhiisol** [comment]
[Fixed](https://github.com/serhiisol/node-decorators/commit/f8ccbc407ee766b74c0d12e0b9fa657f4dbcda45)

**Comment by serhiisol** [comment]
Fixed in [3.4.1](https://github.com/serhiisol/node-decorators/commit/f8ccbc407ee766b74c0d12e0b9fa657f4dbcda45)
